pkg/users/accounts-list.js

```diff
@@ -18,72 +18,277 @@
+    const userGroupLabels = userGroups.map(group => {
+        const color = group.isAdmin ? "gold" : "cyan";
+        return (
+            <Label key={group.name} variant="filled" color={color}>
+                {group.name}
+            </Label>
+        );
+    });
+
+    let loginText = "";
+    let loginSortKey = null;
+    if (account.loggedIn) {
+        loginText = _("Logged in");
+        loginSortKey = "logged in";
+    } else if (!account.lastLogin) {
+        loginText = _("Never logged in");
+        loginSortKey = "never";
+    } else {
+        loginSortKey = new Date(account.lastLogin);
+        loginText = timeformat.dateTime(loginSortKey);
     }
 
-    return (
-        <Card onClick={click} onKeyPress={click} isSelectable isCompact>
-            <CardBody className="cockpit-account">
-                <UserIcon className="cockpit-account-pic" />
-                <div className="cockpit-account-real-name">{account.gecos.split(',')[0]}</div>
-                <div className="cockpit-account-user-name">
+    const columns = [
+        {
+            title: (
+                <span>
                     <a href={"#/" + account.name}>{account.name}</a>
-                    {current && <Badge className="cockpit-account-badge">{_("Your account")}</Badge>}
-                </div>
-            </CardBody>
-        </Card>
-    );
-}
+                    {current && <Badge className="pf-u-ml-lg" id="current-account-badge">{_("Your account")}</Badge>}
+                </span>
+            ),
+            sortKey: account.name,
+            props: { width: 25, },
+        },
+        {
+            title: account.gecos.split(',')[0],
+            props: { width: 20, },
+        },
+        {
+            title: account.uid.toString(),
+            props: { width: 10, },
+        },
+        {
+            title: loginText,
+            sortKey: loginSortKey,
+            props: { width: 25, },
+        },
+        {
+            title: (
+                <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+                    {userGroupLabels}
+                </Flex>
+            ),
+            props: { width: 20, className: "th-align-right" },
+        },
+        {
+            title: <UserActions account={account} />,
+        },
+    ];
 
-export function AccountsList({ accounts, current_user }) {
-    const filtered_accounts = accounts.filter(function(account) {
-        return !((account.uid < 1000 && account.uid !== 0) ||
-                 account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
-                 account.shell === '/bin/false');
-    });
+    return { columns };
+};
+
+const mapGroupsToAccount = (accounts, groups) => {
+    return accounts.map(account => {
+        const accountGroups = [];
+        groups.forEach(group => {
+            if (group.userlist.find(accountName => accountName === account.name))
+                accountGroups.push(group.name);
+        });
+        account.groups = accountGroups;
```

Comment:
Just for the learning process :) Just mentioning the super nice 'reduce' function which can be used like this:
```  diff
diff --git pkg/users/accounts-list.js pkg/users/accounts-list.js
index 17307fa71..bc47d2c95 100644
--- pkg/users/accounts-list.js
+++ pkg/users/accounts-list.js
@@ -80,14 +80,15 @@ const UserActions = ({ account }) => {
 
 const getAccountRow = (account, current, groups) => {
     const admins = ['sudo', 'root', 'wheel'];
-    const userGroups = [];
 
-    groups.forEach((group) => {
+    const userGroups = groups.reduce((pV, group) => {
         if (group.userlist.find(accountName => accountName === account.name)) {
             const isAdmin = !!admins.find(adm => adm === group.name);
-            userGroups.push({ name: group.name, members: group.userlist.length, isAdmin: isAdmin });
+            return pV.concat({ name: group.name, members: group.userlist.length, isAdmin: isAdmin });
+        } else {
+            return pV;
         }
-    });
+    }, []);
```

Reply:
Thanks! I like it!